backend/app/controllers/application_controller.rb

```diff
@@ -1,2 +1,17 @@
 class ApplicationController < ActionController::API
+  rescue_from ActiveRecord::RecordNotFound, ActiveRecord::InvalidForeignKey,
+              ActiveRecord::RecordInvalid, ActiveRecord::Rollback,
+              ActionController::ParameterMissing do |e|
+    render_error({ message: e.message }, :not_found)
```

Comment:
sorry now i confused you: i think this was the only message rendering `:not_found` while all others could be merged and return `render_error({ message: e.message }, :unprocessable_entity)`